src/main/java/io/fabric8/forge/generator/pipeline/ChoosePipelineStep.java

```diff
@@ -298,15 +309,19 @@ public Result execute(UIExecutionContext context) throws Exception {
                         if (Strings.isNullOrBlank(pipelineText)) {
                             status.warning(LOG, "Cannot copy the pipeline to the project as no pipeline text could be loaded!");
                         } else {
-                            if(overrideJenkinsFile.getValue() == true) {
+                            if((isImportRepositoryFlow(attributeMap) && overrideJenkinsFile.getValue() == true)
+                                    || isQuickstartFlow(attributeMap)) {
+                                // overrrideJenkinsFile is null for quickstart wizard flow
+                                // the user in import wizard flow has not opt out for the override
                                 File newFile = new File(basedir, ProjectConfigs.LOCAL_FLOW_FILE_NAME);
                                 Files.writeToFile(newFile, pipelineText.getBytes());
                                 LOG.debug("Written Jenkinsfile to " + newFile);
                             }
                         }
                     }
                 }
-                if(overrideJenkinsFile.getValue() == true) {
+                if((isImportRepositoryFlow(attributeMap) && overrideJenkinsFile.getValue() == true)
```

Comment:
`overrideJenkinsFile.getValue() == true` isn't `true` here redundant?

Reply:
not redundant, we need to check this boolean as the user can opt to not override the jenkins file in the case of import flow only.
but maybe you meant i could just remove `== true`, I guess yes but i find it clearer that way.